src/Components/Admin/AssetModelViewer/assetModelViewer.js

```diff
@@ -0,0 +1,143 @@
+    },
+    textfields: {
+        "& .MuiTextField-root": {
+            margin: theme.spacing(2),
+            width: 205,
+        },
+    },
+}));
+
+export default function AssetModelViewer({
+    match: {
+        params: { assetId },
+    },
+}) {
+    const classes = useStyles();
+    const handleError = useErrorHandler();
+    const [asset, setAsset] = useState({});
+    const [name, setName] = useState("");
+    const [objectType, setObjectType] = useState("");
+
+    useEffect(() => {
+        console.log("using effect");
+        const getObjectAsset = async () => {
+            const data = await getAsset(assetId, handleError);
+            setAsset(data);
+        };
+
+        if (assetId) {
+            getObjectAsset();
+        }
+    }, [assetId, handleError]);
+
+    async function updateAsset(event) {
+        event.preventDefault();
+        console.log(asset);
+        console.log(asset.name);
+        console.log(asset.obj_type);
+
+        const s3_key = asset.s3_key;
+
+        editAsset({ assetId, name, objectType, s3_key }, handleError);
+    }
+
+    return (
+        <div className={classes.root}>
+            <CssBaseline />
+            <Drawer
+                className={classes.drawer}
+                variant="permanent"
+                classes={{
+                    paper: classes.drawerPaper,
+                }}
+                anchor="left"
+            >
+                <div className={classes.toolbar} />
+                <Divider />
+
+                <form
+                    className={classes.textfields}
+                    noValidate
+                    onSubmit={updateAsset}
+                    autoComplete="off"
+                >
+                    <Typography variant="h6" align="center">
+                        Asset Model Viewer
+                    </Typography>
+                    <TextField
+                        id="outlined-helperText"
+                        label="Asset Name"
+                        defaultValue={asset.name}
+                        onChange={(e) => setName(e.target.value)}
+                        variant="outlined"
+                    />
+
+                    <TextField
+                        id="outlined-helperText"
+                        label="Object Type"
+                        defaultValue={asset.obj_type}
+                        onChange={(e) => setObjectType(e.target.value)}
+                        variant="outlined"
+                    />
+
+                    <Button
+                        type="submit"
+                        fullWidth
+                        variant="contained"
+                        color="primary"
+                    >
+                        Save
+                    </Button>
+                </form>
+            </Drawer>
+            <main className={classes.content}>
+                <div className={classes.toolbar} />
+                <model-viewer
+                    src="Astronaut.gltf"
```

Comment:
I feel like this will check for Astronaut in the `public/` folder, not in the place the file actually exists. Might want to try importing the Astronaut file at the top of this file and then using that variable? Not sure if this is even the issue you are facing but just wanted to comment and let you know.

Reply:
Thanks yea, that was actually originally what was causing a problem, I was able to get the glb model showing up in the model viewer

Reply:
Exciting stuff. rest of the code in this PR looked good. Will review and test more closely when you push the working version :)